rgbd/src/client_rgbd.cpp

```diff
@@ -0,0 +1,57 @@
+#include "rgbd/client_rgbd.h"
+
+#include "rgbd/types.h"
+#include "rgbd/ros/conversions.h"
+
+namespace rgbd {
+
+// ----------------------------------------------------------------------------------------
+
+ClientRGBD::ClientRGBD()
+{
+}
+
+// ----------------------------------------------------------------------------------------
+
+ClientRGBD::~ClientRGBD()
+{
+}
+
+// ----------------------------------------------------------------------------------------
+
+void ClientRGBD::intialize(const std::string& server_name)
+{
+    ros::SubscribeOptions sub_options =
+            ros::SubscribeOptions::create<rgbd_msgs::RGBD>(
+                server_name, 1, boost::bind(&ClientRGBD::rgbdImageCallback, this, _1), ros::VoidPtr(), &cb_queue_);
+
+    sub_image_ = nh_.subscribe(sub_options);
+}
+
+// ----------------------------------------------------------------------------------------
+
+bool ClientRGBD::nextImage(Image& image)
+{
+    new_image_ = false;
```

Comment:
I don't like this approach (`new_image_`) as a member bool very much but cannot think of a decent alternative either...

Reply:
One possibility would be to put `cb_queue_`, `image_ptr_` and `new_image_` in a separate class with only a single 'get' method. This, however, is a bit far-fetched.